src/Route.elm

```diff
@@ -19,30 +31,32 @@ type Route
     | Options (Maybe String) (Maybe String) (Maybe String) (Maybe Int) (Maybe Int) (Maybe String)
 
 
+{-| Decode from url with browser search support
+-}
+querySearchString : String -> Url.Parser.Query.Parser (Maybe SearchQuery)
+querySearchString =
+    Url.Parser.Query.map (Maybe.map (SearchQuery << String.replace "+" " "))
+        << Url.Parser.Query.string
+
+
 parser : Url.Parser.Parser (Route -> msg) msg
 parser =
     Url.Parser.oneOf
-        [ Url.Parser.map
-            Home
-            Url.Parser.top
-        , Url.Parser.map
-            NotFound
-            (Url.Parser.s "not-found")
-        , Url.Parser.map
-            Packages
+        [ Url.Parser.map Home Url.Parser.top
+        , Url.Parser.map NotFound (Url.Parser.s "not-found")
+        , Url.Parser.map Packages
             (Url.Parser.s "packages"
                 <?> Url.Parser.Query.string "channel"
-                <?> Url.Parser.Query.string "query"
+                <?> Url.Parser.Query.map (Maybe.map unSearchQuery) (querySearchString "query")
```

Comment:
for now unwrapping inner string directly here so type of routes doesn't need to change.